include/pthread.h

```diff
@@ -0,0 +1,376 @@
+#ifndef _PTHREAD_H
+#define _PTHREAD_H
+
+/* FIXME: This should probably live somewhere else but Zephyr doesn't
+ * currently have a stdc layer to provide it
+ */
+struct timespec {
+	s32_t tv_sec;
+	s32_t tv_nsec;
+};
+
+static inline s32_t _ts_to_ms(const struct timespec *to)
+{
+	return (to->tv_sec * 1000) + (to->tv_nsec / 1000000);
+}
+
+typedef struct pthread_mutex {
+	struct k_sem *sem;
```

Comment:
The k_mutex is indeed a priority-inheriting lock.  It's much heavier weight and not really approprite for a default mutex (POSIX actually allows the implementation to provide both via settings on pthread_mutex_attr -- I didn't write that).  Priority inheritance IMHO has a really limited scope of utility.  The second you have non-lock blocking (e.g. on a semaphore or condition variable) it becomes impossible to implement correctly because those abstrations don't have an "owner" whose priority can be inherited.